### PR TITLE
Refine arguments' types for error interceptors

### PR DESCRIPTION
**Pull request by xamgore**
That PR is an improved solution for #2553.

### CODE REVIEW

**Comment by ahuglajbclajep** [comment]
`AxiosError` now accepts type arguments in #1738.
Therefore, it may be better to set `AxiosError` to `AxiosError<V>`.

**Comment by xamgore** [comment]
@ahuglajbclajep wow, thanks! I have squashed all commits into one.

**Comment by xamgore** [comment]
@Alanscut @chinesedfan could you please review the PR?

**Comment by forgo** [comment]
What happened to this?

**Comment by barakplasma** [comment]
doesn't the return type of the interceptor have to be the same error?

File: index.d.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export interface CancelTokenSource {
 }
 
 export interface AxiosInterceptorManager<V> {
-  use(onFulfilled?: (value: V) => V | Promise<V>, onRejected?: (error: any) => any): number;
+  use(onFulfilled?: (value: V) => V | Promise<V>, onRejected?: (error: AxiosError<V> | any) => any): number;
```
Comment (barakplasma) [inline index.d.ts:123]:
```suggestion
  use(onFulfilled?: (value: V) => V | Promise<V>, onRejected?: (error: AxiosError<V> | any) => Promise<any>): number;
```

**Review by barakplasma — APPROVED** [review]
looks good to me

**Review by barakplasma — CHANGES_REQUESTED** [review]
I think we should make one more little change while we're working on this.

**Review by barakplasma — APPROVED** [review]
Great

**Comment by jasonsaayman** [comment]
Can you please check the merge config? If not I will close this for now and we can reopen.

**Pull request closed without merging**